### ISSUE TITLE
Godot 4 3d stutter in exclusive fullscreen mode

### ISSUE THREAD

**Issue by Jordancjb** [godotengine/godot#90733]
### Tested versions

- Tested in godot stable 4.2.1

### System information

Windows 11, Ryzen 5600x, Radeon RX6750XT, Vulkan, Forward+, Godot 4.2.1

### Issue description

Godot 4 seems to stutter pretty consistently in exclusive fullscreen mode. I first noticed this issue in a project I was working on, where it still happens around half the time I run it or more, making my project unplayable, but it also appears in [this gameplay video of buckshot roulette](https://www.youtube.com/watch?v=zaWPFbX7Q8A&t=289s) most noticable at the 2:20 mark when picking up the shotgun. I'm currently making a project now and this issue was happening very bad in my loading screen and test scene, until I stopped using exclusive fullscreen mode. my project runs fine now, just with an annoying border which has been addressed here before.

### Steps to reproduce

Make a new project, and enable exclusive fullscreen. May need to have some meshes in scene to

### Minimal reproduction project (MRP)

N/A

**Comment by fire** [comment]
3d stutters may be improved in Godot Engine 4.4 https://github.com/godotengine/godot/pull/90400

**Comment by Jordancjb** [comment]
> @Jordancjb Is the stuttering something that only happens the first time you encounter an effect or material in a project's runtime? If so, it's very likely to be shader/PSO compilation stutter.

No, it seems to last for the entire time the scene is run atleast. But I do feel like I've only seen it happen in games where I use high res pbr materials, so I'm not sure if that's part of the problem.

**Comment by Calinou** [comment]
> But I do feel like I've only seen it happen in games where I use high res pbr materials, so I'm not sure if that's part of the problem.

What's the VRAM usage like when running the project? Which screen resolution are you targeting when playing? It's possible that you're running out of VRAM.

I recommend using something like [RTSS](https://www.guru3d.com/download/rtss-rivatuner-statistics-server-download/) or [Special K](https://www.special-k.info/) to check as opposed to the Windows task manager.

**Comment by swAAn01** [comment]
Sorry to necro the thread but did anyone find a solution or root cause for this? I'm having the same issue in my Godot 4.4.1 project.

Here's my setup: 1440p monitor, project settings have window size at 2560x1440

Mode: Windowed runs perfectly fine, no stutters, I'm able to hit a consistent 240 fps

Mode: Fullscreen or Exclusive Fullscreen: game turns into a stuttery mess. I've tried disabling settings one by one, I eventually took out all of the lights, the WorldEnvironment, the VoxelGI, everything I could think of out and it's still a mess, even when I drop resolution

**Comment by fire** [comment]
Check if it’s related to vsync / gsync.

**Comment by swAAn01** [comment]
That is EXACTLY what it was. Thanks!!
